GUI/src/cz/fidentis/gui/NavigatorTopComponent.java

```diff
@@ -480,4 +492,30 @@ public void clearSelectionIfNeeded(ProjectTopComponent project) {
             }
         }
     }
+
+    /**
+     * Creates a copy of given list of FacialPoints reversely transformed by given
+     * transformations. This is needed when displaying original FacialPoints after
+     * models were already registered by the transformations.
+     * @param points facial points of a model
+     * @param trans transformations used to the model
+     * @return copy of facial points reversely transformed by transformations.
+     */
+    private List<FacialPoint> applyReverseICPTrans(List<FacialPoint> points, List<ICPTransformation> trans) {
+        ArrayList<FacialPoint> result = new ArrayList<>(points.size());
+        ArrayList<Vector3f> positions = new ArrayList<>(points.size());
+        
+        // make a copy of points and also store their positions
+        for(int i=0;i<points.size();i++) {
+            FacialPoint point = new FacialPoint(points.get(i));
+            
+            result.add(point);
+            positions.add(point.getPosition());
+        }
+        
+        // apply reverse transformations to positions of points
+        Icp.instance().reverseAllTransformations(trans, positions, true);

```

Comment:
Posledny parameter urcite nebude by default true, vezmi hodnotu z datoveho modelu, malo by sa to vola fpScaling abo cosik poodobne